### PR TITLE
✨ Basic integration with Twilio

### PR DESCRIPTION
**Pull request by dvjones89**
CAST have requested that we build support for the Twilio SMS provider,
in addition to the existing SMS integrations with MMG and Firetext.

This PR introduces the core `TwilioClient` with the ability to send SMS
messages via Twilio's API, instrument success and performance using
`statsd` and, in the event of an unrecoverable error, raising a
Twilio-specific exception populated with meaningful data.

https://app.asana.com/0/1199707043388412/1199945098442987

### CODE REVIEW

**Review by myxrome — APPROVED** [review]

File: app/clients/sms/twilio.py [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+
+from app.clients.sms import (SmsClient, SmsClientResponseException)
+from twilio.rest import Client
+from twilio.base.exceptions import TwilioRestException
+
+class TwilioUtils:
+
+    @staticmethod
+    def get_error_from_code(error_code):
+        with open('app/clients/sms/twilio_error_codes.json') as f:
```
Comment (matthewford) [inline app/clients/sms/twilio.py:12]: this file is huge, is there a way to trim it down to just the error codes for programmable SMS?

Comment (dvjones89) [reply app/clients/sms/twilio.py:12]: @matthewford I tried to find an SMS-only version of the file but don't think it exists. I agree the file is huge, though `python` consumes it (almost) instantaneously.

**Review by matthewford — COMMENTED** [review]
looks good to me, but loading that errors file into ram will increase the ram -it would be good to see if we can trim it down a bit, if not then we'll just need to ensure we have enough headroom.

**Comment by dvjones89** [comment]
Agreed on the `.json` file being RAM intensive, @matthewford. I don't think Twilio author an SMS specific error file, though I'll have another look around the docs and, if needed, will try to find a way of isolating the SMS-specific error codes manually.

**Pull request merged**